src/ilx/EraseUnions.fs

```diff
@@ -71,7 +69,6 @@ type UnionReprDecisions<'Union,'Alt,'Type>
 #if FLATTEN_SINGLE_NON_NULLARY_AND_ALWAYS_USE_TAGS
                 nullPermitted cu then
 #else
-                alts.Length < TaggingThresholdFixedConstant &&
```

Comment:
sry, i didnt understood correctly this part about `RuntimeTypes` vs `IntegerTag`, thx for explaination @dsyme.

i think this pr implements 2)

About performance, i think is more important to use the contract of `UseNullAsTrueValue`, 
because now when i use the attribute i expect null but it doesnt work. 
This attribute is a contract, is not an compiler implementation details

Implementation may change between compiler versions affecting performance (we hope to be always faster), that's ok i think.
We can document this change in the changelog notes. 
`UseNullAsTrueValue` and `> 4 cases`? Remove it for old behavior, leave for use null but check possible performance regression.

With new version, if performance is not ok (the user can see that as regression), the user can remove the `UseNullAsTrueValueAttribute`, moving back to current behaviour (now explicit instead of hidden), using `IntegerTag` and without null value ( that's how it work now).

the 1) can be a future optimization